src/cow-react/modules/swap/pure/EthFlow/EthFlowStepper/index.cosmos.tsx

```diff
@@ -350,6 +313,47 @@ const STEPS: Step[] = [
         state: SmartOrderStatus.CREATION_MINED,
         isExpired: true,
       },
+      creation: { ...defaultProps.creation, failed: false },
+    },
+  },
+  {
+    description: '[TX-FAILED-CREATION] Creation tx failed',
+    props: {
+      ...defaultProps,
+      order: {
+        ...defaultOrderProps,
+        state: SmartOrderStatus.CREATING,
+        isExpired: false,
+        rejectedReason: 'Transaction failed',
+      },
+      creation: { ...defaultProps.creation, failed: true },
+    },
+  },
+  {
+    description: '[TX-FAILED-CANCELLATION] Cancellation tx failed - refund succeeded',
+    props: {
+      ...defaultProps,
+      order: {
+        ...defaultOrderProps,
+        state: SmartOrderStatus.INDEXED,
+        isExpired: false,
+      },
+      creation: { ...defaultProps.creation, failed: false },
+      cancellation: { hash: TX, failed: true },
+      refund: { failed: false },
```

Comment:
you have me lost with this state. Why does it say "xDAI Refunded?"
If I read the state, the order is still OPEN

<img width="1641" alt="image" src="https://user-images.githubusercontent.com/2352112/209008884-3e669c0b-506f-4cb0-b7fc-000a978c8683.png">

Reply:
`refund: {failed: `false`}`

That means the refund succeeded. Maybe I need to update the state to `cancelled` or `expired`

Reply:
Or make it so the cancellation failed and the order is filled